### PR TITLE
Worker Connector Docs

### PR DESCRIPTION
**Pull request by jamiebrynes7**
Updated docs to accompany #981.

Please skip the API docs commit - this is just autogenerated!

### CODE REVIEW

**Comment by jamiebrynes7** [comment]
See preview - http://172.16.121.13:8080/reference/1.0/reference/workflows/monobehaviour/worker-connectors

File: docs/reference/workflows/monobehaviour/worker-connectors.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+# MonoBehaviours: Worker connectors
+
+<%(Callout message="
+Before reading this document, make sure you have read:
+
+  * [Workers in the GDK]({{urlRoot}}/reference/concepts/worker)
+  * [Connection flows]({{urlRoot}}/reference/concepts/connection-flows)
+")%>
+
+To expose the [`Worker`]({{urlRoot}}/api/core/worker) class in the MonoBehaviour workflow, the GDK provide an abstract [`WorkerConnector`]({{urlRoot}}/api/core/worker-connector) MonoBehaviour.
```
Comment (paulbalaji) [inline docs/reference/workflows/monobehaviour/worker-connectors.md]: I'd explain what the worker connector is first before mentioning you can access the Worker through it (flip paragraphs + smooth out to flow)

File: docs/reference/workflows/monobehaviour/worker-connectors.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+    protected override void HandleWorkerConnectionFailure()
+    {
+        Debug.Log("Failed to create worker.");
+    }
+
+    public override void Dispose()
+    {
+        // Need to call base.Dispose() to properly clean up.
+        // The ECS world and the connection.
```
Comment (paulbalaji) [inline docs/reference/workflows/monobehaviour/worker-connectors.md]:
```suggestion
        // the ECS world and the connection.
```

File: docs/reference/workflows/monobehaviour/worker-connectors.md [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    }
+
+    protected override void HandleWorkerConnectionFailure()
+    {
+        Debug.Log("Failed to create worker.");
+    }
+
+    public override void Dispose()
+    {
+        // Need to call base.Dispose() to properly clean up.
```
Comment (paulbalaji) [inline docs/reference/workflows/monobehaviour/worker-connectors.md]:
```suggestion
        // Need to call base.Dispose() to properly clean up
```

File: docs/reference/workflows/monobehaviour/worker-connectors.md [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+        base.Dispose();
+    }
+}
+```
+
+<%(/Expandable)%>
+
+## How to modify the connection configuration
+
+When configuring the connection, you have a number of parameters to configure. The GDK provides a small framework for composing these parameters which you can extend to build reusable configurations.
```
Comment (paulbalaji) [inline docs/reference/workflows/monobehaviour/worker-connectors.md]:
```suggestion
There are a number of parameters to consider when configuring the connection. The GDK provides a small framework for composing these parameters, which you can then extend to build reusable configurations.
```

File: docs/reference/workflows/monobehaviour/worker-connectors.md [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+
+The `SpatialOSConnectionHandlerBuilder` object is central to this framework, it allows you to declare which connection flow to use and what parameters the connection should use. There are two key decisions that you need to make when configuring your connection.
```
Comment (paulbalaji) [inline docs/reference/workflows/monobehaviour/worker-connectors.md]:
```suggestion
The `SpatialOSConnectionHandlerBuilder` object is central to this framework. It allows you to declare which connection flow to use, what parameters the connection should use, and how these parameters are populated.
```

File: docs/reference/workflows/monobehaviour/worker-connectors.md [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+
+### What connection parameters should I use
+
+The [`WorkerConnector`]({{urlRoot}}/api/core/worker-connector) class has a `CreateConnectionParameters(string workerType)` method which returns reasonable defaults for your `ConnectionParameters`. You can then tweak these to fit your use case:
```
Comment (paulbalaji) [inline docs/reference/workflows/monobehaviour/worker-connectors.md]: maybe? idm either way `...your use case. For example: `

File: docs/reference/workflows/monobehaviour/worker-connectors.md [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+
+- If you are implementing a client worker, you may want to use the `KCP` networking stack.
+- If you are debugging the low level networking of a worker, you may want to enable [protocol logging](https://docs.improbable.io/reference/latest/shared/debugging#protocol-logging).
+
+### How do I populate my connection flow parameters
+
+Each of the [connection flows]({{urlRoot}}/reference/concepts/connection-flows) have parmeters that need to be configured to ensure a successful connection.
```
Comment (paulbalaji) [inline docs/reference/workflows/monobehaviour/worker-connectors.md]:
```suggestion
Each of the [connection flows]({{urlRoot}}/reference/concepts/connection-flows) have parameters that need to be configured to ensure a successful connection.
```

File: docs/reference/workflows/monobehaviour/worker-connectors.md [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+- `CommandLineConnectionFlowInitializer` which can initialize the `ReceptionistFlow,` `LocatorFlow`, and `AlphaLocatorFlow` from command line parameters.
+- `MobileConnectionFlowInitializer` which can initialize the `ReceptionistFlow` and `AlphaLocatorFlow`.
+
+## How do I customize the connection flows
+
+The Locator and AlphaLocator connection flows both have callbacks to the user during the connection process. The `LocatorFlow` and `AlphaLocatorFlow` objects provide a reasonable default implementation for these callbacks, but you may want to customize this to suit your connection & authentication logic. 
+
+To customize these flows, simply inherit from the relevant flow object and override the methods that you want to tweak. Please refer to the API documentation for more detail on the [`LocatorFlow`]({{urlRoot}}/api/core/locator-flow) and [`AlphaLocatorFlow`]({{urlRoot}}/api/core/alpha-locator-flow).
+
+
```
Comment (paulbalaji) [inline docs/reference/workflows/monobehaviour/worker-connectors.md]: nit: line

File: docs/reference/concepts/worker.md [hunk 9]
```diff
@@ -33,7 +33,7 @@ To learn about a workers’ connection to the SpatialOS Runtime, see:
 
 To learn more about how to create and use workers in the GDK, please see the following documentation:
 
-  * MonoBehaviour workflow - [Workers: Creating workers with WorkerConnector]({{urlRoot}}/reference/workflows/monobehaviour/creating-workers)
+  * MonoBehaviour workflow - [Workers: Creating workers with WorkerConnector]({{urlRoot}}/reference/workflows/monobehaviour/worker-connectors)
```
Comment (paulbalaji) [inline docs/reference/concepts/worker.md]:
```suggestion
  * MonoBehaviour workflow - [Workers: Creating workers with the WorkerConnector]({{urlRoot}}/reference/workflows/monobehaviour/worker-connectors)
```

Comment (jamiebrynes7) [reply docs/reference/concepts/worker.md]: How long has that been there..

File: docs/reference/workflows/monobehaviour/worker-connectors.md (hunk 1 above)
Comment (jamiebrynes7) [reply docs/reference/workflows/monobehaviour/worker-connectors.md]: I guess the intention is you've read about the `Worker` class (noted in the box at the top) - now lets segue into the `WorkerConnector` 🤔

File: docs/reference/workflows/monobehaviour/worker-connectors.md (hunk 6 above)
Comment (jamiebrynes7) [reply docs/reference/workflows/monobehaviour/worker-connectors.md]: Yeah I think I prefer how you've written it

**Pull request merged**